cupy/cuda/compiler.py

```diff
@@ -580,7 +586,8 @@ def _compile_with_cache_cuda(
         base = _preprocess('', options, arch, backend)
         _empty_file_preprocess_cache[env] = base
 
-    key_src = '%s %s %s %s' % (env, base, source, extra_source)
+    key_src = '%s %s %s %s %s' % (
+        env, base, source, extra_source, _get_cupy_cache_key())
```

Comment:
Technically we don't need `extra_source` here anymore, because `CUPY_CACHE_KEY` is a superset of it (hash of all 1st-party & 3rd-party headers). However I decided to do minimal work and leave it here for now. Perhaps we could revisit this and clean up the `extra_source` related code later?